### DISCUSSION TITLE
Construct thb spline problem

### DISCUSSION THREAD

**Discussion by Rennngw** [gismo/gismo#586] [Q&A] [answered]
Hello, i use a gsTensorBSpline to construct gsTHBSpline. There is something weird happened.
Is it because this gsTensorBSpline is not suitable for constructing gsTHBSpline？
I don't know why.
My code and picture as follows:
```
int main()
{
std::string geo_path = "plateWithHole.xml";
gsGeometry<>::uPtr mpPtr = gsReadFile<>(geo_path);
gsTensorBSplineBasis<2, real_t>::uPtr mbaseptr = gsReadFile<>(geo_path);
auto& patch = *mpPtr;
auto& tensorbas = *mbaseptr;
gsTensorBSpline<2, double> bSpline(tensorbas,patch.coefs());
gsTHBSpline<2, double> thbSpline(bSpline);
gsWriteParaview(bSpline, "bSpline", 1000, 1, 1);
gsWriteParaview(thbSpline, "thbSpline", 1000, 1, 1);
}
```

gsTensorBSpline is normal:
![image](https://user-images.githubusercontent.com/110526351/218096354-892b1545-3a84-4661-9c6d-d5ee5abc49e4.png)

gsTHBSpline is not normal:
![image](https://user-images.githubusercontent.com/110526351/218096483-1e9b1c71-3473-43ee-8ffb-bb0c810133fd.png)

Thannks in advance.

**Comment by Rennngw** [comment]
[plateWithHole.zip](https://github.com/gismo/gismo/files/10707504/plateWithHole.zip)

**Comment by ckarampa** [comment]
Hello, we have decided on using the Q&A section of the Github Discussions for questions, instead of the Issues page ( that should be reserved for bugs, etc. ). Since Discussions don't vanish as they are closed, we thought this a good way to have an archive of questions that have been addressed, for anyone reading in the future. This is why I am converting the issues you created into discussions. Please ask your future questions in this manner. 

Regarding your questions, I have not used THB splines yet, so I am not the best person to help you, but I'll have a closer look once I find time.

**Comment by ckarampa** [comment]
Tried replicating your problem ( copy pasted your code snippet ) and on my system it works as expected, bspline.pvd and thbspline.pvd are identical in ParaView. What system are you on? Did you check that it is not a paraview malfunction ( tried restarting it etc. ) ?

**Reply by Rennngw** [reply]
Sorry for not answering earlier and ok i will ask my future questions in this manner.
My system is windows and i restart my paraview, it still has the same problem.
I have tried many methods, such as initializing Hbspline, but the result is still abnormal.
I don't know why.
Is it because we are using different versions of gismo？

The following is the function call history of my initialization thbspline：

1. `gsTHBSpline<2, double> thbSpline(bSpline);`
2. 
![image](https://user-images.githubusercontent.com/110526351/218233906-743b0b67-e98b-47b1-b941-24f2ba9e646b.png)
3.
![image](https://user-images.githubusercontent.com/110526351/218233938-b40493ce-ea4f-485c-ae77-60b79c87b040.png)

Does my function call history differ from yours？If we are the same, why do I have this problem, I have been troubled by this problem for a long time, thank you very much for your help.
Thank you very much for your reply in advance.

Best regards
Rennngw

**Reply by ckarampa** [reply]
Same call history here.. I really don't know what it might be. You said that 
> Is it because we are using different versions of gismo？

Have you modified gismo? Maybe you have accidentally changed something that causes it. Try running it with gismo/stable

**Comment by Rennngw** [comment, marked as answer]
Ok, thank you very much for your reply.
Running it with gismo/stable is success.
Thank you for your help again.
